sdk/couchbase-core/src/queryx/query_options.rs

```diff
@@ -523,3 +523,344 @@ impl Serialize for QueryOptions {
+        self.ignore_if_not_exists = ignore_if_not_exists.into();
+        self
+    }
+
+    pub fn on_behalf_of(mut self, on_behalf_of: impl Into<Option<OnBehalfOfInfo>>) -> Self {
+        self.on_behalf_of = on_behalf_of.into();
+        self
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+#[non_exhaustive]
+pub struct DropIndexOptions {
+    pub(crate) bucket_name: String,
+    pub(crate) scope_name: Option<String>,
+    pub(crate) collection_name: Option<String>,
+    pub(crate) index_name: String,
+    pub(crate) ignore_if_not_exists: Option<bool>,
+    pub(crate) on_behalf_of: Option<OnBehalfOfInfo>,
+}
+
+impl DropIndexOptions {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn bucket_name(mut self, bucket_name: impl Into<String>) -> Self {
+        self.bucket_name = bucket_name.into();
+        self
+    }
+
+    pub fn scope_name(mut self, scope_name: impl Into<Option<String>>) -> Self {
+        self.scope_name = scope_name.into();
+        self
+    }
+
+    pub fn collection_name(mut self, collection_name: impl Into<Option<String>>) -> Self {
+        self.collection_name = collection_name.into();
+        self
+    }
+
+    pub fn index_name(mut self, index_name: impl Into<String>) -> Self {
+        self.index_name = index_name.into();
+        self
+    }
+
+    pub fn ignore_if_not_exists(mut self, ignore_if_not_exists: impl Into<Option<bool>>) -> Self {
+        self.ignore_if_not_exists = ignore_if_not_exists.into();
+        self
+    }
+
+    pub fn on_behalf_of(mut self, on_behalf_of: impl Into<Option<OnBehalfOfInfo>>) -> Self {
+        self.on_behalf_of = on_behalf_of.into();
+        self
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+#[non_exhaustive]
+pub struct BuildDeferredIndexesOptions {
+    pub(crate) bucket_name: String,
+    pub(crate) scope_name: Option<String>,
+    pub(crate) collection_name: Option<String>,
+    pub(crate) on_behalf_of: Option<OnBehalfOfInfo>,
+}
+
+impl BuildDeferredIndexesOptions {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn bucket_name(mut self, bucket_name: impl Into<String>) -> Self {
+        self.bucket_name = bucket_name.into();
+        self
+    }
+
+    pub fn scope_name(mut self, scope_name: impl Into<Option<String>>) -> Self {
+        self.scope_name = scope_name.into();
+        self
+    }
+
+    pub fn collection_name(mut self, collection_name: impl Into<Option<String>>) -> Self {
+        self.collection_name = collection_name.into();
+        self
+    }
+
+    pub fn on_behalf_of(mut self, on_behalf_of: impl Into<Option<OnBehalfOfInfo>>) -> Self {
+        self.on_behalf_of = on_behalf_of.into();
+        self
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+#[non_exhaustive]
+pub struct WatchIndexesOptions {
+    pub(crate) bucket_name: String,
+    pub(crate) scope_name: Option<String>,
+    pub(crate) collection_name: Option<String>,
+    pub(crate) timeout: Duration,
```

Comment:
I think this timeout probably shouldn't exist. The watch operation will just run until the future is dropped.